File: docs/snippets/go/benchmarking/simple_benchmark.go
```go
package main

import (
	"fmt"
	"sync"
	"time"

	"kreuzberg"
)

func main() {
	config := &kreuzberg.ExtractionConfig{
		UseCache: false,
	}
	client, err := kreuzberg.New(config)
	if err != nil {
		panic(err)
	}
	defer client.Close()

	filePath := "document.pdf"
	numRuns := 10

	fmt.Printf("Sync extraction (%d runs):\n", numRuns)
	start := time.Now()
	for i := 0; i < numRuns; i++ {
		_, err := client.ExtractFile(filePath)
		if err != nil {
			panic(err)
		}
	}
	syncDuration := time.Since(start).Seconds()
	avgSync := syncDuration / float64(numRuns)
	fmt.Printf("  - Total time: %.3fs\n", syncDuration)
	fmt.Printf("  - Average: %.3fs per extraction\n", avgSync)

	fmt.Printf("\nAsync extraction (%d parallel runs):\n", numRuns)
	start = time.Now()
	var wg sync.WaitGroup
	wg.Add(numRuns)
	for i := 0; i < numRuns; i++ {
		go func() {
			defer wg.Done()
			_, err := client.ExtractFile(filePath)
			if err != nil {
				panic(err)
			}
		}()
	}
	wg.Wait()
	asyncDuration := time.Since(start).Seconds()
	fmt.Printf("  - Total time: %.3fs\n", asyncDuration)
	fmt.Printf("  - Average: %.3fs per extraction\n", asyncDuration/float64(numRuns))
	fmt.Printf("  - Speedup: %.1fx\n", syncDuration/asyncDuration)

	cacheConfig := &kreuzberg.ExtractionConfig{
		UseCache: true,
	}
	clientCached, err := kreuzberg.New(cacheConfig)
	if err != nil {
		panic(err)
	}
	defer clientCached.Close()

	fmt.Println("\nFirst extraction (populates cache)...")
	start = time.Now()
	_, err = clientCached.ExtractFile(filePath)
	if err != nil {
		panic(err)
	}
	firstDuration := time.Since(start).Seconds()
	fmt.Printf("  - Time: %.3fs\n", firstDuration)

	fmt.Println("Second extraction (from cache)...")
	start = time.Now()
	_, err = clientCached.ExtractFile(filePath)
	if err != nil {
		panic(err)
	}
	cachedDuration := time.Since(start).Seconds()
	fmt.Printf("  - Time: %.3fs\n", cachedDuration)
	fmt.Printf("  - Cache speedup: %.1fx\n", firstDuration/cachedDuration)
}
```

File: docs/snippets/rust/benchmarking/simple_benchmark.rs
```rust
use kreuzberg::{Kreuzberg, ExtractionConfig, CacheConfig};
use std::time::Instant;

#[tokio::main]
async fn main() -> anyhow::Result<()> {
    let config = ExtractionConfig {
        use_cache: false,
        ..Default::default()
    };
    let kreuzberg = Kreuzberg::new(config)?;
    let file_path = "document.pdf";
    let num_runs = 10;

    let start = Instant::now();
    for _ in 0..num_runs {
        let _ = kreuzberg.extract_file(file_path).await?;
    }
    let sync_duration = start.elapsed().as_secs_f64();
    let avg_sync = sync_duration / num_runs as f64;

    println!("Sync extraction ({} runs):", num_runs);
    println!("  - Total time: {:.3}s", sync_duration);
    println!("  - Average: {:.3}s per extraction", avg_sync);

    let start = Instant::now();
    let mut tasks = vec![];
    for _ in 0..num_runs {
        tasks.push(kreuzberg.extract_file(file_path));
    }
    let results = futures::future::join_all(tasks).await;
    for result in results {
        result?;
    }
    let async_duration = start.elapsed().as_secs_f64();

    println!("\nAsync extraction ({} parallel runs):", num_runs);
    println!("  - Total time: {:.3}s", async_duration);
    println!("  - Average: {:.3}s per extraction", async_duration / num_runs as f64);
    println!("  - Speedup: {:.1}x", sync_duration / async_duration);

    let cache_config = ExtractionConfig {
        use_cache: true,
        ..Default::default()
    };
    let kreuzberg_cached = Kreuzberg::new(cache_config)?;

    println!("\nFirst extraction (populates cache)...");
    let start = Instant::now();
    let result1 = kreuzberg_cached.extract_file(file_path).await?;
    let first_duration = start.elapsed().as_secs_f64();
    println!("  - Time: {:.3}s", first_duration);

    println!("Second extraction (from cache)...");
    let start = Instant::now();
    let result2 = kreuzberg_cached.extract_file(file_path).await?;
    let cached_duration = start.elapsed().as_secs_f64();
    println!("  - Time: {:.3}s", cached_duration);
    println!("  - Cache speedup: {:.1}x", first_duration / cached_duration);

    Ok(())
}
```
